### PR TITLE
Generate and test complex infras

### LINKED ISSUE #255: Add more complex infrastructures and run tests on it

### PR DESCRIPTION
**Pull request by eckter**
Fixes #255

### CODE REVIEW

File: examples/generator/railjson_generator/schema/infra/endpoint.py [hunk 1]
```diff
@@ -23,6 +23,12 @@ def get_neighbors(self) -> "TrackEndpoint":
             return self.track_section.begining_links
         return self.track_section.end_links
 
+    def opposite(self) -> "TrackEndpoint":
+        if self.endpoint is Endpoint.BEGIN:
+            return TrackEndpoint(self.track_section, Endpoint.END)
+        else:
+            return TrackEndpoint(self.track_section, Endpoint.BEGIN)
```
Comment (flomonster) [inline examples/generator/railjson_generator/schema/infra/endpoint.py]:
```suggestion
        return TrackEndpoint(self.track_section, Endpoint.BEGIN)
```

File: examples/generator/railjson_generator/schema/infra/infra.py [hunk 2]
```diff
@@ -20,6 +20,7 @@ class Infra:
     operational_points: List[OperationalPoint] = field(default_factory=list)
     tvd_sections: List[TVDSection] = field(default_factory=list)
     routes: List[Route] = field(default_factory=list)
+    speed_limits: List[Dict] = field(default_factory=list)
```
Comment (flomonster) [inline examples/generator/railjson_generator/schema/infra/infra.py]: Do we need to specify the label of a speed limit, we could generate it from speed limits in track sections.

File: examples/generator/railjson_generator/schema/infra/track_section.py [hunk 3]
```diff
@@ -29,6 +29,9 @@ class TrackSection:
     end_coordinates: Optional[Tuple[float, float]] = field(default=None)
     begining_links: List[TrackEndpoint] = field(default_factory=list, repr=False)
     end_links: List[TrackEndpoint] = field(default_factory=list, repr=False)
+    slopes: List[Tuple[float, float, float]] = field(default_factory=list)
+    curves: List[Tuple[float, float, float]] = field(default_factory=list)
+    speed_limits: List[Tuple[float, float, str]] = field(default_factory=list)
```
Comment (flomonster) [inline examples/generator/railjson_generator/schema/infra/track_section.py]: I would have added multiple datastruct such as `RangeElement`, `Slope`, `Curve` and `SpeedLimit`  to stay consistent with the rest of the library.

File: examples/generator/railjson_generator/schema/infra/track_section.py [hunk 4]
```diff
@@ -91,8 +103,21 @@ def format(self):
...
+            "curves": [{
+                "begin": begin,
+                "end": end,
+                "radius": radius,
+            } for begin, end, radius in self.curves],
+            "speed_sections": [{
+                "begin": begin,
+                "end": end,
+                "ref": section_id,
+                "applicable_direction": "BOTH"
```
Comment (flomonster) [inline examples/generator/railjson_generator/schema/infra/track_section.py]: All of this could be simplify by calling a `format()` method

**Review by flomonster — CHANGES_REQUESTED** [review]

**Review by bgiuliana — APPROVED** [review]

**Review by flomonster — APPROVED** [review]

**Pull request merged**